bait-files/src/main/java/com/google/sps/servlets/UsernameTakenServlet.java

```diff
@@ -41,20 +33,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String username = request.getParameter("username");
 
     response.setContentType("application/json");
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     if (username == null || username.equals("")) {
         String json = new Gson().toJson(true);
         response.getWriter().println(json);
         return;
     }
 
-    Query query = new Query("User")
-      .setFilter(new Query.FilterPredicate(
-        "username", Query.FilterOperator.EQUAL, username));
-    PreparedQuery results = datastore.prepare(query);
-    Entity entity = results.asSingleEntity();
-    boolean usernameTaken = entity != null; 
+    List<User> allUsers = ObjectifyService.ofy().load().type(User.class).list();
 
+    List<User> usersWithUsername =
+        allUsers.stream().filter(user -> user.username.equals(username)).collect(toList());        
+        
+    boolean usernameTaken = usersWithUsername.size() >= 1; 
```

Comment:
This logic looks identical to that the of the userIsRegistered method in LoginFilter.  It's only a few lines long so you don't have to yet... but if you find yourself duplicating much more common logic like this, you may want to extract it out into a new helper class meant for executing common user functionality =)

Reply:
I definitely will if it gets longer